src/Aspire.Hosting/Dcp/DcpExecutorEvents.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Concurrent;
+using Aspire.Hosting.ApplicationModel;
+
+namespace Aspire.Hosting.Dcp;
+
+internal record ResourceStatus(string? State, DateTime? StartupTimestamp, DateTime? FinishedTimestamp);
```

Comment:
Why didn't you use an interface here? Do we have multiple subscribers?

Reply:
There are never multiple subscribers for these events. Just ApplicationOrchestrator uses this to listen to changes from DCP.

Are you asking about why not an interface for `DcpExecutorEvents`? There is only one implementation. The type could disappear and it becomes part of `DcpExecutor`/`IDcpExecutor`, but having it separated feels more usable in tests.

Reply:
I’m asking why isn’t the dcp executor just calling interface methods.